### ISSUE TITLE
mobifyjs or adaptivejs

### ISSUE THREAD

**Issue by shystruk** [mobify/mobifyjs#300]
Hello!

I have found framework Adaptive.js on Mobify.com.
Is the mobifyjs older product and doesn't use now ?
Better use Adaptive.js ?

Thank you!

**Comment by jansepar** [comment]
Hi @shystruk!

Adaptive.js is our new framework for adapting the experience of existing websites for difference devices. Mobify.js (1.X) was initially also a framework for adapting the experience of existing websites, but has evolved (in 2.0) to become more of a core library for a number of Mobify services. Mobify.js is actually a dependancy in Adaptive.js!

Here is a high level overview of the differences between mobify.js 1.1 and adaptive.js:

Uses modern toolchains:
- Require.js for dependancy management (rather then using templating to bring in different libraries - a feature most people likely didn't even touch in mobify.js).
- Grunt.js for the task runner. This replaces `mobify-client` that you'd typically use with mobify.js projects. Adaptive.js comes with a base set of Grunt plugins, but you're free to add any other plugins/tasks into your project that you wish (imagine for example, you wanted to run your tests on Saucelabs - that is very easy to integrate into your project using the provided Gruntfile).
- Bower for package management of front-end assets. While it was possible to use bower with mobify.js, we make it easy to use with adaptive.js

More modular:
- With mobify.js, it was very difficult to bring in different libraries into your project when doing the initial adaptation (the konf). Since adaptive.js uses require.js at it's core, its very easy to swap libraries, such as which selector engine you'd like to use (jquery vs zepto), which templating engine you'd like to use (although dust.js is highly encouraged), etc.

Uses an MVC paradigm:
- With mobify.js, all of your controller and routing logic is in one giant file - the mobify.konf. In adaptive.js, we've moved things to be much more MVC-like to make things more familiar for developers. For example, the controller logic for the header, footer, and each view are contained within their own files. The router is also a file of its own that maps routes to views. We find it to be much cleaner then the "context.choose" method of mobify.js.

Easier to debug:
- With mobify.js, debugging inline scripts in your templates was difficult, and unfortunately, writing inline scripts was encouraged. With adaptive.js, we start people off right by putting those scripts in an external file for easy debugging, as well as to reduce the filesize of the initially loaded file for improved performance.

Improved performance:
- There are many performance tweaks we made that makes adaptive.js shine over mobify.js. And stay tuned, there are plenty more improvements to come!

Testable:
- Writing tests for mobify.js was very difficult and not built into the framework. Adaptive.js was built with testing in mind right from the get go. When you generate an adaptive.js project, it comes with a set of basic tests to start you off on a road of writing amazing tests to ensure your project is stable and reliable!

Hopefully that answers your questions!
